head/src/main/kotlin/io/qalipsis/core/head/lock/LockProvider.kt

```diff
@@ -0,0 +1,52 @@
+/*
+ * QALIPSIS
+ * Copyright (C) 2022 AERIS IT Solutions GmbH
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+package io.qalipsis.core.head.lock
+
+import io.qalipsis.api.context.CampaignKey
+
+/**
+ * Provides methods for lock management including creation, removal and access.
+ *
+ * @author Francisca Eze
+ */
+internal interface LockProvider {
+
+    /**
+     * Creates an instance of the lock for the provided campaign key.
+     *
+     * @param campaignKey reference of the running campaign
+     */
+    suspend fun createLock(campaignKey: CampaignKey) = Unit
```

Comment:
Remove, this will be replaced by the campaign hook functions on the implementations.